ws-nextjs-app/jest.config.ts

```diff
@@ -1,7 +1,35 @@
 import { pathsToModuleNameMapper } from 'ts-jest/utils';
+import type { Config } from '@jest/types';
 import { compilerOptions } from '../tsconfig.json';
 
-const config: import('jest').Config = {
+const canonicalIntegrationTests = {
+  displayName: 'Integration Tests - Canonical',
+  testEnvironment: './integration/IntegrationTestEnvironment',
+  testEnvironmentOptions: {
+    platform: 'canonical',
+  },
+  setupFilesAfterEnv: ['./setupTests.ts'],
+  transform: {
+    '^.+\\.(js|jsx|ts|tsx)$': ['babel-jest', { presets: ['next/babel'] }],
+  },
+  testMatch: ['**/integration/!(utils)/**/*[^.amp].test.js'],
+} as Config.InitialProjectOptions;
+
+const ampIntegrationTests = {
```

Comment:
Do we need AMP for our NextJS app?

Reply:
Good question, I think we will for frontpages and articles in the future. I may need to ask Neil if we have an expectation to support amp on live pages.

Reply:
I guess if the eventuality is moving everything to Next then it's probably fine leaving this testing pattern in, but yeah would be good to know in general

Reply:
We don't need amp for live pages, but we will for other page types, does this setup support selective support for amp?

Reply:
Cool, changed it so that the int tests only cover the canonical ones, and then added a yarn command 'test:integration:amp' for the amp test suits specifically.